### ISSUE TITLE
Anwenderfrage: zwei XTD übergeben?

### ISSUE THREAD

**Issue by Pierre-de-la-Verre** [claeis/ilivalidator#382]
Vielleicht banal ..

Auf der Seite des VSA  findet sich dieser Befehlsaufruf:
`java -jar ilivalidator-1.14.0.jar –allObjectsAccessible –log transferdatensatz_2020_d_LV95_T-20201109.log transferdatensatz_2020_d_LV95_T-20201109.xtf vsa_organisationen_testdatensatz.xtf`

Wenn ich das richtig verstehe wird damit wird der "transferdatensatz" geprüft unter der Hinzunahme von "vsa_organisationen". Klappt bei mir auch - nur warum? Besser gesagt - wo finde ich die Dokumentation dazu? Habe ich das übersehen?

**Comment by edigonzales** [comment]
Ja, das ist so. Ich habe auch nichts Explizites in der Doku gefunden. Insbesondere in Kombination mit --allObjectsAccessible.

**Comment by Pierre-de-la-Verre** [comment]
Danke.
Jetzt frage ich mich natürlich:
- Ist das nur genau so: 1 "Daten-XTF" plus optional ein "Adress-XTF"?
- oder plus mehrere Adress-XTF?
- oder plus irgendwas?
Als (Unter?)Durchschnittsanwender schwimmt man da schon ein wenig ..

**Comment by edigonzales** [comment]
Die Magie beginnt beim Modell. Als Beispiel unser Nutzungsplanungsmodell https://geo.so.ch/models/ARP/SO_Nutzungsplanung_20171118.ili Es hat mehrere Topics. Darunter das Topic "Rechtsvorschriften" und das Topic "Nutzungsplanung". In den "Rechtsvorschriften" werden die Rechtsvorschriften erfasst, in der "Nutzungsplanung" die eigentlichen (Geo-)Daten. Zwischen Geodaten und Rechtsvorschriften gibt es eine Assoziation, so à la "Welche Rechtsvorschrift gilt für diese Geometrie?". Weil die jeweiligen Objekte in unterschiedlichen Topics sind muss eine Rolle in der Assoziation als EXTERNAL definiert werden:

```
    ASSOCIATION Typ_Grundnutzung_Dokument =
      Typ_Grundnutzung -- {0..*} Typ_Grundnutzung;
      Dokument (EXTERNAL) -- {0..*} SO_Nutzungsplanung_20171118.Rechtsvorschriften.Dokument;
    END Typ_Grundnutzung_Dokument;
```

Rechtsvorschriften und Nutzungsplanung müssen mit dieser Modellierung nicht zwingend in der gleichen Datei geliefert werden. Bei uns werden jedoch die Rechtsvorschriften immer mitgeliefert. Wenn ich aber mit ilivalidator eine XTF-Datei prüfe, wird standardmässig die Assoziation nicht geprüft. Weil die Rechtsvorschriften eben nicht zwingend vorhanden sein müssen. Dazu gibt es den Schalter "--allObjectsAccessible". Damit teilt man ilivalidator mit, dass in der zu prüfenden Datei (vermeintlich) alle Objekte vorhanden sind und auch EXTERNAL Rollen geprüft werden sollen.

Das ganze lässt sich nun kombinieren mit der Möglichkeit mehrere Dateien gleichzeitig zu prüfen. Beim VSA-DSS-mini ist es leicht anders modelliert, aber im Kern ist es das gleiche. Du hast die Organisationen in einer anderen Datei und hast jedoch eine Beziehung zwischen Objekten der (Geo-)Daten und den Organisationen. Wenn du in deinem Fall --allObjectsAccessible nicht verwendest, werden die beiden XTF separat validiert. 

Es gibt noch eine weitere Magie: Du muss das Organisations-XTF gar nicht vorliegen haben, sondern ilivalidator findet es automatisch:

```
java -jar ilivalidator.jar --allObjectsAccessible ilidata:vsa_organisationen geodaten.xtf
```

Das bedingt, dass die Organisationen in einem INTERLIS-Repo vorliegen: https://vsa.ch/models/ilidata.xml Darum das Kürzel "ilidata". Dann durchsucht ilivalidator die bekannten Repos nach der Datei mit der ID "vsa_organisationen" und lädt sie herunter.

**Comment by edigonzales** [comment]
Ah und am 22. November findet in Olten das dritte INTERLIS-Anwender:innen-Treffen statt: https://interlis.discourse.group/t/3-interlis-anwender-innen-treffen-save-the-date/128

Wahrscheinlich werden wird am Nachmittag Workshops zum Thema "Bessere Daten" durchführen. D.h. es geht um die Validierung der Daten. Vom Normalo über den Fortgeschrittenen bis zum Freak-Level. (je 1 WS).

**Comment by beistehen** [comment]
Review IUG: @claeis Doku ergänzt in PR #453, würde damit geschlossen
